lib/db/mysql.js

```diff
@@ -281,7 +281,16 @@ module.exports = function (log, error) {
     return this.readFirstResult(
       CHECK_PASSWORD,
       [uid, hash.verifyHash]
-    )
+    ).then(function(result) {
+      return P.resolve(result)
+    }, function(err) {
+      // If .readFirstResult() doesn't find anything, it returns a error.notFound()
+      // so we need to convert that to an error.incorrectPassword()
+      if ( err.code === error.notFound().code && err.errno === error.notFound().errno ) {
```

Comment:
Having to create an instance here seems awkward.  Can we extract these into named constants, either here or in the `error` module?  Perhap that would require a larger refactor of the `error` module to avoid duplicate, in which case feel free to defer to a follow-up bug.  I'm thinking something like:

```
err.is(error.notFound)
```

In the meantime, the `errno` uniquely identifies an error so IMHO we don't need to bother with checking the code.